packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -268,14 +268,7 @@ export async function fetchServerResponse(
     // If fetch fails handle it like a mpa navigation
     // TODO-APP: Add a test for the case where a CORS request fails, e.g. external url redirect coming from the response.
     // See https://github.com/vercel/next.js/issues/43605#issuecomment-1451617521 for a reproduction.
-    return {
-      flightData: url.toString(),
-      canonicalUrl: undefined,
-      couldBeIntercepted: false,
-      prerendered: false,
-      postponed: false,
-      staleTime: -1,
-    }
+    return doMpaNavigation(url.toString())
```

Comment:
`doMpaNavigation`:
https://github.com/vercel/next.js/blob/5ae7baa0e8dea2528c7b6e07d774ec729c5aa8cb/packages/next/src/client/components/router-reducer/fetch-server-response.ts#L82C1-L94C1

actual diff is just following:

```diff
-     flightData: url.toString(),
+     flightData: urlToUrlWithoutFlightMarker(url).toString(),
```